src/gui/properties/speedplotview.cpp

```diff
@@ -207,67 +239,53 @@ void SpeedPlotView::setGraphEnable(GraphID id, bool enable)
 
 void SpeedPlotView::pushPoint(const SpeedPlotView::PointData &point)
 {
-    m_data5Min.push_back(point);
-    m_averager30Min.push(point);
-    m_averager6Hour.push(point);
-    m_averager12Hour.push(point);
-    m_averager24Hour.push(point);
+    for (Averager *averager : {&m_averager5Min, &m_averager30Min
+                                , &m_averager6Hour, &m_averager12Hour
+                                , &m_averager24Hour})
+    {
+        // SPOILER ALERT: conditions are short circuited, make sure push comes first
+        if (averager->push(point) && m_currentData == averager)
+            viewport()->update();
+    }
 }
 
 void SpeedPlotView::setPeriod(const TimePeriod period)
 {
-    m_period = period;
-
     switch (period)
     {
     case SpeedPlotView::MIN1:
-        m_viewablePointsCount = MIN1_SEC;
-        m_currentData = &m_data5Min;
+        m_averager5Min.setCurrentMaxDuration(DURATION_MIN1);
```

Comment:
Wouldn't it be easier to have separate averagers for 1 and 5 min graphs? Is there a real gain in sharing the same averager, but at the same time complicating it internally?

Reply:
no, this doesn't complicate anything imo, only place where it has any effect is this https://github.com/qbittorrent/qBittorrent/pull/14054/files#diff-989e95031bfc96747860213bce0a55b9cf1d2175cbe3de0db42190945398ba91R180

but that equation will be required anyway, plus since this PR improves the resolution for the 6-hour graph So, I was thinking of re-using the 6-hour averager to add a 3-hour graph by using `Averager::setCurrentMaxDuration`

Reply:
>no, this doesn't complicate anything imo

Really? The presence of both `max duration` and `current max duration` in averager logic, makes it confusing. But most importantly, what is the advantage of this approach? Why not just have a dedicated averager for each used interval?

Reply:
> Really? The presence of both max duration and current max duration in averager logic, makes it confusing. But most importantly, what is the advantage of this approach? Why not just have a dedicated averager for each used interval

With this you don't need to have another array for storing the values, saving memory, and then you don't have to run the average on every stat update, saving cpu resources.

Reply:
>With this you don't need to have another array for storing the values, saving memory, and then you don't have to run the average on every stat update, saving cpu resources.

Why then, based on the same assumptions, not have only one Averager? What is the line that separates these cases?

Reply:
> Why then, based on the same assumptions, not have only one Averager? What is the line that separates these cases?

resolution (after how much duration Averager averages) https://github.com/qbittorrent/qBittorrent/pull/14054/files#diff-989e95031bfc96747860213bce0a55b9cf1d2175cbe3de0db42190945398ba91R57 this increases with the duration of the target time period, it's a trick to decrease the memory usage. So 24-hour averager will accumulate data till the elapsed time < resolution {144s) and the average it and push it in the data queue https://github.com/qbittorrent/qBittorrent/pull/14054/files#diff-989e95031bfc96747860213bce0a55b9cf1d2175cbe3de0db42190945398ba91R135